app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,123 @@
+ * Activity for camera preview and taking photos
+ */
+public class CameraActivity extends AppCompatActivity {
+
+    private CameraView camera;
+    private boolean mCapturingPicture;
+
+    /**
+     * onCreate method of the Android Activity Lifecycle
+     * @param savedInstanceState the bundle of the last instance state saved
+     * @author Bedin Elia
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_camera);
+        camera = findViewById(R.id.viewCamera);
+
+        //interface method for automatic onResume, onPause, onDestroy
+        camera.setLifecycleOwner(this);
+
+        //Pinch to zoom support
+        camera.mapGesture(Gesture.PINCH, GestureAction.ZOOM);
+        //Tap to focus support
+        camera.mapGesture(Gesture.TAP, GestureAction.FOCUS_WITH_MARKER);
+        //Long tap to capture support (currently disabled, should be .CAPTURE)
+        camera.mapGesture(Gesture.LONG_TAP, GestureAction.NONE);
+
+        camera.addCameraListener(new CameraListener() {
+            /**
+             * Listining for photo taken by the camera
+             * @param jpeg the taken photo in array form
+             * @author Bedin Elia
+             */
+            public void onPictureTaken(byte[] jpeg) { onPicture(jpeg); }
+        });
+
+        ImageButton buttonPhoto = findViewById(R.id.capturePhoto);
+        buttonPhoto.setOnClickListener(new View.OnClickListener() {
+            /**
+             * Listening for button click, then call capturePhoto() to take a photo
+             * @param view the layout components
+             * @author Bedin Elia
+             */
+            @Override
+            public void onClick(View view) {
+                capturePhoto();
+            }
+        });
+
+    }
+
+    /**
+     * Method for taking a photo
+     * @author Bedin Elia
+     */
+    private void capturePhoto() {
+        if (mCapturingPicture) return;
+        mCapturingPicture = true;
+        camera.capturePicture();
+    }
+
+    /**
+     * Method for sending the taken photo in array form to the PicturePreviewActivity
+     * @param jpeg the taken photo
+     * @author Bedin Elia
+     */
+    private void onPicture(byte[] jpeg) {
+        mCapturingPicture = false;
+        //sending the taken photo to PicturePreviewActivity via setImage
+        PicturePreviewActivity.setImage(jpeg);
+        //sending information to PicturePreviewActivity through Intent
+        Intent intent = new Intent(CameraActivity.this, PicturePreviewActivity.class);
+        startActivity(intent);
+    }
+
+    /**
+     * Permission checker for Android M or higher
+     * @param requestCode permission code
+     * @param permissions the requested permissions
+     * @param grantResults the grant results for the corresponding permissions
+     * @author Bedin Elia
+     */
+    @Override
+    public void onRequestPermissionsResult
```

Comment:
I have not clear where in the code there is the checking of the permissions that eventually launch this method.
Accordingly to this https://developer.android.com/training/permissions/requesting#perm-check it should be similar to this:
`if (ContextCompat.checkSelfPermission(CameraActivity.this, Manifest.permission.CAMERA)`
`        != PackageManager.PERMISSION_GRANTED) {`
`    ActivityCompat.requestPermissions(CameraActivity.this,` 
`                new String[]{Manifest.permission.CAMERA},`
`                MY_PERMISSIONS_REQUEST_CAMERA);`
`}`
Have I understood your intentions? Is it included in the library so it's not necessary?

Reply:
The library already do the permission for Camera, but it's implicit like Android K and lower, when there was no promp to ask you.
This code it's for having a visual prompt, and it's an override for the current library method.

Reply:
mmm, are you sure? Maybe I'm not understand. Is onRequestPermissionResult for showing a visual prompt for the permissions? https://developer.android.com/reference/android/support/v4/app/ActivityCompat#requestpermissions here it's not seem to be so. In that link I've read that the ActivityCompat.requestPermission launch the visual prompt to the user and the onRequestPermissionsResult catches the results of the request.
But if it is already implemented in the library it's ok.

Reply:
Yeah, the library does, see https://github.com/natario1/CameraView#declaration.
Thanks again.

Reply:
Ok, thanks to you for this specifications. I've left that point of the documentation! Thanks to you.

Reply:
No problem!